src/scripts/import-needs-assessment-data/add-regions.ts

```diff
@@ -0,0 +1,241 @@
+      }
+      return resultsMap;
+    }, resultsMap);
+
+  console.log("Add Geo.Regions results:");
+  Object.keys(resultsMap).forEach((key) => {
+    console.log(`     ${key}: ${resultsMap[key].length}`);
+
+    // NOTE: uncomment & set the status key to debug different types of results
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validRegions: Region[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((regions: Region[], workflow: RegionUploadWorkflow) => {
+    return [...regions, workflow.data];
+  }, [] as Region[]);
+
+  return validRegions;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/*  Consolidate Regions
+ * ------------------------------------------------------- */
+function consolidateRegions(data: NeedAssessment[]): string[] {
+  const regions = data.reduce((acc: string[], item) => {
+    const region = item.place?.region;
+    if (region && !acc.includes(region)) {
+      acc.push(region);
+    }
+    return acc;
+  }, []);
+
+  return regions;
+}
+
+/*  Parse Region
+ * ------------------------------------------------------- */
+function parseRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): RegionUploadWorkflow {
+  logs = [...logs, `Log: parsing region "${orig}"`];
+
+  if (orig == null || typeof orig !== "string") {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: Invalid region input: "${orig}". Expected a non-null value.`,
+      ],
+    };
+  }
+
+  const newData = {
+    ...data,
+    region: orig,
+  };
+
+  return {
+    data: newData,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/*  Get Region
+ * ------------------------------------------------------- */
+async function getRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): Promise<RegionUploadWorkflow> {
+  logs = [...logs, `Log: Checking if Geo.Region "${orig}" already exists.`];
+
+  //Fetch the data from Strapi
+  const response = await fetch(`${STRAPI_ENV.URL}/regions?`, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${STRAPI_ENV.KEY}`,
+    },
+  });
+
+  const body = await response.json();
+  const matchingRegion = body.data.find(
+    (region) => region.Name.toLowerCase() === data.region.toLowerCase(),
+  );
+
+  if (!response.ok) {
```

Comment:
Can the two if (!response.ok) errors with just one word different be refactored and combined? And I'd look for other places to deduplicate code and send in just the entity name where identical things are happening in multiple places.